app/src/main/java/io/intrepid/contest/screens/contestjudging/EntryViewHolder.java

```diff
@@ -8,24 +9,36 @@
 
 import butterknife.BindView;
 import butterknife.ButterKnife;
+import butterknife.OnClick;
+import butterknife.Optional;
 import io.intrepid.contest.R;
 import io.intrepid.contest.models.Entry;
 
 class EntryViewHolder extends RecyclerView.ViewHolder {
+    private final EntryOnClickListener listener;
     @BindView(R.id.entry_submission_title_field)
     TextView submissionTitleField;
+    private Entry entry;
 
-    EntryViewHolder(ViewGroup parent) {
-        super(inflateView(parent));
+    EntryViewHolder(ViewGroup parent, @LayoutRes int rowLayout, EntryOnClickListener listener) {
+        super(inflateView(parent, rowLayout));
         ButterKnife.bind(this, itemView);
+        this.listener = listener;
     }
 
-    private static View inflateView(ViewGroup parent) {
+    private static View inflateView(ViewGroup parent, @LayoutRes int rowLayout) {
         LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
-        return layoutInflater.inflate(R.layout.entry_submission_row, parent, false);
+        return layoutInflater.inflate(rowLayout, parent, false);
+    }
+
+    @Optional
```

Comment:
Yes if I hard-code one res layout per VH class, I don't need any checks like this. The problems come when I attempt to use one vh to support multiple row types depending on the screen